src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,86 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75F;
+    private int size;
+    private Node<K,V>[] table;
+
+    public MyHashMap() {
+        table = new Node[INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int index = hash(key, table.length);
+        Node<K,V> node = table[index];
+        if (table[index] == null) {
+            table[index] = new Node<>(key, value, null);
+            size++;
+            checkForResize();
+            return;
+        }
+        while (node != null) {
+            if (node.key == key || (node.key != null
+                    && node.key.equals(key))) {
+                node.value = value;
+                return;
+            }
+            node = node.next;
+        }
+        Node<K,V> newNode = new Node<>(key,value,table[index]);
+        table[index] = newNode;
+        size++;
+        checkForResize();
     }
 
     @Override
     public V getValue(K key) {
+        Node<K,V> value = table[hash(key,table.length)];
+        while (value != null) {
+            if (value.key == key || (value.key != null
+                    && value.key.equals(key))) {
+                return value.value;
+            }
+            value = value.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void checkForResize() {
```

Comment:
bad logic inside this method. Now it doesn't move values from one bucket to other. For example, if you have only 2 not empty bucket firstly, most likely, you will have 4 not empty bucket after resize. But now you get the same 2 not empty bucket. You need to resize in appropriate way to decrease collision effect. I would recommend to think about one loop, set size to 0 and reusing method `put()`